internal/controller/scenario/scenario_controller.go

```diff
@@ -84,6 +92,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	})
 	if err != nil {
 		logger.Error(err, "Failed to get Application from the IntegrationTestScenario after retry", "application", scenario.Spec.Application)
+		return ctrl.Result{}, nil
```

Comment:
Once we have retried long enough for the Application to exist in the cluster, and still couldn't find it, the operator will stop processing this ITS. But in that case, this ITS would be an orphan and dangle in the cluster. Do you think we should delete this ITS here?

Reply:
Hm, not sure, maybe we can at least mark it as invalid?